packages/react-ui/src/app/routes/flows/index.tsx

```diff
@@ -186,15 +190,65 @@ const FlowsPage = () => {
             table.getIsAllPageRowsSelected() ||
             (table.getIsSomePageRowsSelected() && 'indeterminate')
           }
-          onCheckedChange={(value) => table.toggleAllPageRowsSelected(!!value)}
-        />
-      ),
-      cell: ({ row }) => (
-        <Checkbox
-          checked={row.getIsSelected()}
-          onCheckedChange={(value) => row.toggleSelected(!!value)}
+          onCheckedChange={(value) => {
+            const isChecked = !!value;
+            table.toggleAllPageRowsSelected(isChecked);
+
+            if (isChecked) {
+              const allRowIds = table
+                .getRowModel()
+                .rows.map((row) => row.original);
+
+              const newSelectedRowIds = [...allRowIds, ...selectedRowIds];
+
+              const uniqueRowIds = Array.from(
+                new Map(
+                  newSelectedRowIds.map((item) => [item.id, item]),
+                ).values(),
+              );
+
+              setSelectedRowIds(uniqueRowIds);
+            } else {
+              const filteredRowIds = selectedRowIds.filter((row) => {
+                return !table
+                  .getRowModel()
+                  .rows.some((r) => r.original.version.id === row.version.id);
+              });
+              setSelectedRowIds(filteredRowIds);
+            }
+          }}
         />
       ),
+      cell: ({ row }) => {
```

Comment:
I will leave this comment here because I can't comment on the place that's broken.
Basically, all places that should refresh the table after an action don't work, setRefresh is not making the table refresh nor the folders list.

Reply:
This bug is also in the connections table, probably in the runs table as well